File: content/examples/other/leetcode/problems/number-of-1-bits/example-2/Solution.java
public class Solution {
  static int hammingWeight(int n) {
    int count = 0;
    while (n != 0) {
      n &= (n - 1);
      count += 1;
    }
    return count;
  }

  public static void main(String[] args) {
    System.out.println(hammingWeight(0b00000000000000000000000000001011));
  }
}

File: content/examples/other/leetcode/problems/number-of-1-bits/example-2/solution.js
function hammingWeight(n) {
  n = n >>> 0;
  let count = 0;
  while (n !== 0) {
    n = (n & (n - 1)) >>> 0;
    count += 1;
  }
  return count;
}

if (require.main === module) {
  console.log(hammingWeight(0b00000000000000000000000000001011));
}

module.exports = { hammingWeight };
